### PR TITLE
Introduce stimulus reflex, adjust variants form

### PR DESCRIPTION
**Pull request by Jbern16**
Variants Form will now:

Update on field change,
Add another.

Accomplishes this using stimulus reflex. Some scaffold stuff to wire this up, including redis which now will start via foreman.

### CODE REVIEW

File: app/controllers/product_variants_controller.rb [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 
   def create
-    @variant = Variant.new(variant_params)
-    if @variant.save
-      redirect_to new_product_image_path(@variant.product)
-    else
-      render :new
-    end
+    # catch errors in save from reflex?
+    @product ||= Product.find_by_id(product_params[:product_id])
```
Comment (pindell-matt) [inline app/controllers/product_variants_controller.rb:9]: for future we could do a before_action to set product, e.g.

```
before_action :set_product, on: [:new, :create]
def set_product
     @product ||= Product.find_by_id(product_params[:product_id])
end
```

File: app/views/product_variants/new.html.erb [hunk 2]
```diff
@@ -3,18 +3,21 @@
     <h3 class="font-bold text-2xl">Tell us a bit more</h3>
   </section>
   <section class="mt-10">
-  <%= form_for([@product, @variant], url: product_variants_path, class: 'flex flex-col') do |f| %>
+  <%= form_with model: @product, method: 'post', url: product_variants_path, class: 'flex flex-col', data: {signed_id: @product.to_sgid.to_s} do |f| %>
```
Comment (pindell-matt) [inline app/views/product_variants/new.html.erb:6]: typo? `signed_id: @product.to_sgid.to_s`

Comment (pindell-matt) [reply app/views/product_variants/new.html.erb:6]: I guess not, I've never seen `.to_sgid` - I'm just a simple country boy

Comment (Jbern16) [reply app/views/product_variants/new.html.erb:6]: I believe it encrypts it to prevent some sort of form injection through action cable

**Review by pindell-matt — APPROVED** [review]
Looks good!

**Pull request merged**